libcudacxx/test/libcudacxx/cuda/uninitialized_buffer/access.pass.cpp

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the libcu++ Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11
+// UNSUPPORTED: msvc-19.16
+// UNSUPPORTED: nvrtc
+#define LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE
+
+#include <cuda/buffer>
+#include <cuda/memory_resource>
+#include <cuda/std/cassert>
+#include <cuda/std/span>
+
+#include <nv/target>
+
+struct do_not_construct
+{
+  do_not_construct()
+  {
+    assert(false);
+  }
+};
+
+template <class T>
+void test()
+{
+  using uninitialized_buffer = cuda::mr::uninitialized_buffer<T>;
+
+  cuda::mr::cuda_memory_resource resource{};
+  const uninitialized_buffer buf{resource, 42};
+  assert(buf.data() != nullptr);
+  assert(buf.size() == 42);
+
+  assert(buf.begin() == buf.data());
+  assert(buf.end() == buf.begin() + buf.size());
+}
```

Comment:
Could you please test writing and reading from the buffer? So the test would crash if the underlying memory was not available. E.g. just `std::iota()` the buffer and `std::reduce` the content.

Reply:
Done